### ISSUE TITLE
401 Client Error: Unauthorized for url

### ISSUE THREAD

**Issue by jskimm** [chainflag/eth-challenge-base#19]
Hi, I'm struggling from deploying `example contract` with your PoA geth private Network.

I got an error as below when I tried to deploy the contract.

```
$ nc 172.x.x.x 20000

We design a pretty easy contract challenge. Enjoy it!
Your goal is to make isSolved() function returns true!

[1] - Create an account which will be used to deploy the challenge contract
[2] - Deploy the challenge contract using your generated account
[3] - Get your flag once you meet the requirement
[4] - Show the contract source code
[-] input your choice: 2
[-] input your token: v4.local.bFdMp90mn5dSUcQGMymwkCr2_TO5tTxc~~
401 Client Error: Unauthorized for url: http://172.x.x.x:8545/
```

### Steps to reproduce:
1. Following the steps in the `/geth` directory to make an account and a genesis block, and running `docker-compose up` to start PoA geth private network.
2. Run `docker run -it -p 20000:20000 -e WEB3_PROVIDER_URI=http://172.x.x.x:8545 chainflag/eth-challenge-base` 
3. Connect to the challenge server like `nc 172.x.x.x 20000`
4. Create an account and request 1ETH in the facuet
5. Select `2` to deploy a new contract and got an error

So I checked the tcp packet and there was an 401 error since `eth_getBlockByNumber` was not whitelisted.

![그림1](https://user-images.githubusercontent.com/7110910/156950137-b695556c-b521-4aab-8851-0f56ac71c41b.png)

Please let me know if I missed anything or if there are any other logs that you need.

**Comment by iczc** [comment]
Thank you for finding this bug. I updated the transaction type of contract deployment from `Legacy` to `EIP1559`,  and the transaction type requires querying the historical blocks to calculate the basefee.  But we disabled the `eth_getBlockByNumber`method in the private chain to prevent players from getting someone else's txhash,  this is the root cause of the 401 error.

I will fix this issue later, for now, you can use the Ethereum testnet as `WEB3_PROVIDER_URI` or use the 0.9.2 version without EIP1559.

**Comment by iczc** [comment]
Resolved by #10. Currently the `main` tag of docker image(`chainflag/eth-challenge-base:main`)includes this PR,  and I'll re-release `0.9.3` in several days.

**Comment by iczc** [comment]
Released
